codalab/rest/interpret.py

```diff
@@ -306,134 +314,86 @@ def resolve_interpreted_items(interpreted_items):
+            'id': item_index,
+            'text': 'ERROR: ' + message,
+        }).data
 
-    for item in interpreted_items:
+
+    for item_index, item in enumerate(interpreted_items):
         if item is None:
             continue
         mode = item['mode']
-        data = item['interpreted']
-        properties = item['properties']
 
         try:
             # Replace data with a resolved version.
-            if mode == 'markup':
+            if mode == BlockModes.markup_block:
                 # no need to do anything
                 pass
-            elif mode == 'record' or mode == 'table':
+            elif mode == BlockModes.record_block or mode == BlockModes.table_block:
                 # header_name_posts is a list of (name, post-processing) pairs.
-                header, contents = data
+                header = item['header']
+                contents = item['rows']
                 # Request information
                 contents = interpret_genpath_table_contents(contents)
-                data = (header, contents)
-            elif mode == 'contents':
-                try:
-                    max_lines = int(properties.get('maxlines', DEFAULT_CONTENTS_MAX_LINES))
-                except ValueError:
-                    raise UsageError("maxlines must be integer")
 
-                target_info = rest_util.get_target_info(data, 0)
-                if target_info is not None and target_info['type'] == 'directory':
-                    data = ['<directory>']
+                item['rows'] = contents
+            elif mode == BlockModes.contents_block or mode == BlockModes.html_block or mode == BlockModes.image_block:
+                target_info = rest_util.get_target_info((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), 0)
+
+                if target_info is not None and target_info['type'] == 'directory' and mode == BlockModes.contents_block:
+                    item['status']['code'] = 'ready'
+                    item['lines'] = ['<directory>']
                 elif target_info is not None and target_info['type'] == 'file':
-                    data = head_target(data, max_lines, replace_non_unicode=True)
-                else:
-                    data = None
-            elif mode == 'html':
-                target_info = rest_util.get_target_info(data, 0)
-                if target_info is not None and target_info['type'] == 'file':
-                    data = head_target(data, None)
+                    item['status']['code'] = 'ready'
+                    if mode == BlockModes.contents_block:
+                        item['lines'] = head_target((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), item['max_lines'], replace_non_unicode=True)
+                    elif mode == BlockModes.html_block:
+                        item['html_lines'] = head_target((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), item['max_lines'])
```

Comment:
html_lines doesn't make much sense since truncated HTML isn't valid.  Probably better to just have `html` and slurp up the entire file (like how we treat images).

I thought we were going to deprecate HTML anyway?

Reply:
Yeah we were going to deprecate HTML, should I do it in this change? was just replicating functionality for this change and then take it out in a separate one.